large update to module

milestorme opened the pull request:
removed all useless and not working enchants, fixed all travis errors and updated calculations to be a little fairer and fixed 3rd enchant slot, added files to db_assembler. Major updates, player tested on my server for a few months now , working well

milestorme commented:
@wizzymore @BarbzYHOOL , can you check this one too pls

BarbzYHOOL commented on src/RandomEnchants.cpp:
```diff
@@ -1,63 +1,139 @@
 			return;
+		}
+
+		// It's what we want..
```
which information does it convey?

BarbzYHOOL commented on src/RandomEnchants.cpp:
```diff
@@ -1,63 +1,139 @@
+bool QuestReward = 1;
+uint32 HighQuality = 5;
+uint32 LowQuality = 1;
+uint32 Chance1 = 60;
```
why define them here? why not directly inline (see other comment below)

BarbzYHOOL replied:
hmm in fact they are already set to default values (60 65 70) directly below in the code

IN FACT, why do you put default values ? the defautl values can stay in the config too because if they are missing, the module will use the .conf.dist (or display error and then the user has to fix it)

BarbzYHOOL commented on src/RandomEnchants.cpp:
```diff
@@ -1,63 +1,139 @@
+		QuestReward = sConfigMgr->GetBoolDefault("RandomEnchants.OnQuestReward", 1);
+		HighQuality = sConfigMgr->GetIntDefault("RandomEnchants.HighQuality", 1);
+		LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
+		Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
```
I was suggesting this
```suggestion
		Chance1 = (sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60)) ? sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60) : 60
```

then i realized the "60" is already there and anyway see my comments above (aka default values should stay in the conf for clarity and no duplicate code)

BarbzYHOOL commented on src/RandomEnchants.cpp:
```diff
@@ -1,63 +1,139 @@
+		LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
+		Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
+		Chance2 = sConfigMgr->GetIntDefault("RandomEnchants.Chance2", 65);
+		Chance3 = sConfigMgr->GetIntDefault("RandomEnchants.Chance3", 70);
```
same here ofc

wizzymore replied:
Please fix the indents here as well

BarbzYHOOL commented on src/RandomEnchants.cpp:
```diff
@@ -1,63 +1,139 @@
+			if (Crafted == true)
+			{
+				RollPossibleEnchant(player, item);
+			}
```
just realized you use tabs instead of space and it looks very empty on github xD

milestorme replied:
ah crap, not sure if it was like this on stygiancore or if i did this, i think i got this module from there originally and updated it , this will require me to rewrite it all :( , maybe i can get to that after christmas

wizzymore replied:
Or you can simply replace all the "	" with "    ". From tab indents to 4 spaces indents, simple 15 seconds fix

BarbzYHOOL commented on src/RandomEnchants.cpp:
```diff
@@ -1,63 +1,139 @@
 		uint32 slotEnch[3] = { 0, 1, 5 };
 		double roll1 = rand_chance();
-		if (roll1 >= 70.0)
+		if (roll1 >= Chance1) //
```
```suggestion
		if (roll1 >= Chance1)
```

? I remove the // dunno what they're for

wizzymore commented on src/RandomEnchants.cpp:
```diff
@@ -1,63 +1,139 @@
+uint32 LowQuality = 1;
+uint32 Chance1 = 60;
+uint32 Chance2 = 65;
+uint32 Chance3 = 70;
```
Define them on a single line and remove the default values

wizzymore commented:
Please fix the indentation and put all the defined unsigned ints for the config on one line with no default value, or on multiple lines but still, with no default value.

I will test the module in the following days, if you fix the above and everything works as it should this could be pulled

milestorme commented:
Ok I’ll do that thank you 🙏

BarbzYHOOL commented on src/RandomEnchants.cpp:
```diff
@@ -1,155 +1,214 @@
+        LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
+        Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
+        Chance2 = sConfigMgr->GetIntDefault("RandomEnchants.Chance2", 65);
+        Chance3 = sConfigMgr->GetIntDefault("RandomEnchants.Chance3", 70);
```
ok so hmm, the defautl values here, they will only happen if they are not found in randomenchants.conf.dist which is very unlikely

so why let them here?

I feel like it's better to remove them from the code and have them only defined in the conf

What do you think @milestorme @wizzymore ? (i know it's a detail)

BarbzYHOOL replied:
milestorme, you resolved the conv but didn't answer lol

milestorme replied:
if You want , doesn’t really matter .

milestorme replied:
@wizzymore see

BarbzYHOOL commented:
for tabs, yeah it's better for us to use 4 spaces instead (note: you can probably enable the plugin for editorconfig in your IDE, it's made for that) since we do that on the core too (and in fact when you do python/yaml you realize that you can't use tabs xD)

milestorme commented:
yeah i know i updated this module soooo long ago on torn before i changed that havnt made this mistake for months since you guys told me way back then i just never merged this one cause it wasnt finished being tested. it should be all fixed now

milestorme commented:
@BarbzYHOOL should be all fixed

wizzymore commented on src/RandomEnchants.cpp:
```diff
@@ -39,16 +39,6 @@ class REConfig : public WorldScript
-        LowQuality = sConfigMgr->GetIntDefault("RandomEnchants.LowQuality", 1);
-        Chance1 = sConfigMgr->GetIntDefault("RandomEnchants.Chance1", 60);
-        Chance2 = sConfigMgr->GetIntDefault("RandomEnchants.Chance2", 65);
-        Chance3 = sConfigMgr->GetIntDefault("RandomEnchants.Chance3", 70);
```
I don't see this variables getting the values from the config in any other place than this, you sure you are supposed to delete this?

milestorme replied:
Absolutely not sure at all but @BarbzYHOOL told me to , who am I argue .

milestorme replied:
Actually I’m done here , they way you treat my freinds is bad , do your own work from now on I’ll only be doing it privately

milestorme replied:
Whe y’all decide to stop being bullies to Poszer and talmortis and I might decide to contribute here again

BarbzYHOOL commented:
@wizzymore i asked above what do you guys think about it but you didn't answer me both

BarbzYHOOL commented:
ok stoabrogga told me it's better to keep the default value for ALL the config options (so modules should work fine even if there is no configuration file found, I thought it was better if they didn't work if config file missing but I was wrong), so the last commit is not wanted, hence why I asked for your opinion guys :P

wizzymore commented:
Yeah, you don't say

Winfidonarleyan commented:
Rofl
